### PR TITLE
Use ld-cheri-elf.so.1 if it exists for purecap.

### PR DESCRIPTION
**Pull request by bsdjhb**
This fixes dynamic binaries in a hybrid world.

### CODE REVIEW

**Comment by bsdjhb** [comment]
I haven't tested it yet, but this is the same fix Jess did for RISC-V.

File: sys/mips/mips/elf_machdep.c [hunk 1]
```diff
@@ -159,11 +159,12 @@ static __ElfN(Brandinfo) freebsd_brand_info = {
...
 	.interp_path	= "/libexec/ld-elf.so.1",
 	.sysvec		= &elf_freebsd_sysvec,
-	.interp_newpath	= NULL,
 #if __has_feature(capabilities)
+	.interp_newpath	= "/libexec/ld-cheri-elf.so.1",
 	.header_supported = mips_elf_header_supported,
 	.flags		= BI_CAN_EXEC_DYN
 #else
+	.interp_newpath	= NULL,
 	.brand_note	= &__elfN(freebsd_brandnote),
```
Comment (jrtc27) [inline sys/mips/mips/elf_machdep.c:168]: Curious that this is ELF64-only; for RISC-V, the brand note stuff has been left as the same for both and things seem to work...

File: sys/mips/mips/elf_machdep.c [hunk 2]
```diff
@@ -159,11 +159,12 @@ static __ElfN(Brandinfo) freebsd_brand_info = {
 	.emul_path	= NULL,
 	.interp_path	= "/libexec/ld-elf.so.1",
 	.sysvec		= &elf_freebsd_sysvec,
-	.interp_newpath	= NULL,
 #if __has_feature(capabilities)
+	.interp_newpath	= "/libexec/ld-cheri-elf.so.1",
 	.header_supported = mips_elf_header_supported,
```
Comment (jrtc27) [inline sys/mips/mips/elf_machdep.c:164]: Since we've removed CHERI256 support this can go

Comment (bsdjhb) [reply sys/mips/mips/elf_machdep.c:164]: It's used for the 'use_cheriabi' check to decide between COMPAT_CHERIABI and the native ABI for purecap.  I've previously removed the 256 bit checks from it.

File: sys/mips/mips/elf_machdep.c (hunk 1 above)
Comment (bsdjhb) [reply sys/mips/mips/elf_machdep.c:168]: I think I had just preserved this from one in sys/mips/cheri/cheriabi_machdep.c.  It probably should work for MIPS as well.

File: sys/mips/mips/elf_machdep.c (hunk 2 above)
Comment (jrtc27) [reply sys/mips/mips/elf_machdep.c:164]: Oh, yeah, that needs to stay. But the calls to (and inline implementation of) cheriabi_check_cpu_compatible are still there.

**Pull request merged**